### ISSUE TITLE
UUID does not match

### ISSUE THREAD

**Issue by 18601673727** [loyd/clickhouse.rs#26]
```rs
use serde::{Deserialize, Serialize};

#[derive(clickhouse::Row, Debug, Serialize, Deserialize)]
pub struct Test {
    #[serde(with = "uuid::serde::compact")]
    pub id: uuid::Uuid,
}

let mut inserter = db.insert("tests")?;
inserter.write(&Test { id: Uuid::new_v4() }).await?; // id: cb5f628b-0f76-4e3e-a310-9b15cb9d29ee
inserter.end().await?; // success

// later in clickhouse-client do:
// select * from tests;

┌─id───────────────────────────────────┐
│ 3e4e760f-8b62-5fcb-ee29-9dcb159b10a3 │
└──────────────────────────────────────┘
```

**Comment by zryambus** [comment]
I have a some tricky solution, but it can be improved:
```rust
#[derive(Debug)]
pub struct UUID(pub uuid::Uuid);

impl From<uuid::Uuid> for UUID {
    fn from(value: uuid::Uuid) -> Self {
        Self { 0: value }
    }
}

fn uuid_to_bytes_be(id: &uuid::Uuid) -> [u8; 16] {
    let bytes = id.as_bytes();
    let mut result: [u8; 16] = [0; 16];
    for i in 0..16 {
        let multiplier = i / 8;
        let j = i % 8;
        let index = 8 * multiplier + 7 - j;
        result[index] = bytes[i];
    }
    result
}

impl Serialize for UUID {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: serde::Serializer
    {
        let bytes = uuid_to_bytes_be(&self.0);
        serde::Serialize::serialize(&bytes, serializer)
    }
}
```
The idea is to move the bytes in the next order
```
Before  0 1 2 3 4 5 6 7   8 9 10 a b c  d e f
After   7 6 5 4 3 2 1 0   f e  d c b a 10 9 8
```

After this you can pass `UUID` object to `write` method of `Inserter` and this should work as expected

**Comment by 18601673727** [comment]
Thanks @zryambus !

**Comment by loyd** [comment]
It obviously must be handled in this library, so I've reopened an issue.
Variants:
* Move to `RowBinaryWithNamesAndTypes` (https://github.com/loyd/clickhouse.rs/issues/10) and detect `UUID` types by CH. However, it works well for deserialization, but serialization requires fetching an actual schema of a table.
* Check a struct name in the (de)serializer. It hurts the performance of all newtypes and has false positives.
* Provide the `uuid` feature and the wrapper around, as above.
* Provide a custom `Uuid` type and conversions to `uuid::Uuid` under the `uuid` feature.

What do you thing about variants?

**Comment by zryambus** [comment]
I think that last two variants are pretty good.
Before #26 will be resolved, it will be great to add to readme the note, that there are some issues with serialization of `uuid::Uuid` type

**Comment by loyd** [comment]
Published in `v0.11.0`
